src/util/hash.ts

```diff
@@ -1,9 +1,47 @@
 /** @module ssz */
-import SHA256 from "@chainsafe/as-sha256";
+import SHA256, {HashObject} from "@chainsafe/as-sha256";
 
 /**
  * Hash used for hashTreeRoot
  */
 export function hash(...inputs: Uint8Array[]): Uint8Array {
   return SHA256.digest(Buffer.concat(inputs));
 }
+
+/** To save memory, we don't want to always allocate a new HashObject */
+const sharedHashObject: HashObject = {
+  h0: 0,
+  h1: 0,
+  h2: 0,
+  h3: 0,
+  h4: 0,
+  h5: 0,
+  h6: 0,
+  h7: 0,
+};
+
+/** Clone a hash object using sharedHashObject */
```

Comment:
Could you add more details in the JSDoc comment about the mechanics of this sharedHashObject? It's not quite clear to me.

Reply:
this is used in a lot of places then we apply the shared HashObject to persistent-merkle-tree, it'll make a copy so it's safe to mutate it after that.
so yes, it will get mutated in different places but it's safe to do so.
I'll add more comments regarding this.